diff --git a/thrift/lib/CMakeLists.txt b/thrift/lib/CMakeLists.txt
index f318c28dc3..127e77fa77 100644
--- a/thrift/lib/CMakeLists.txt
+++ b/thrift/lib/CMakeLists.txt
@@ -224,10 +224,11 @@ if(thriftpy3)
     WORKING_DIRECTORY ${_cybld}
   )
   add_custom_command(TARGET thrift_python_types_bindings
-    COMMAND
-      CXX="${CMAKE_CXX_COMPILER}"
-      LDSHARED="${CMAKE_CXX_COMPILER} -shared"
-      python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py --api-only
+    COMMAND ${CMAKE_COMMAND} -E env
+      "CFLAGS=${CMAKE_C_FLAGS}"
+      "CXXFLAGS=${CMAKE_CXX_FLAGS}"
+      "CXX=${CMAKE_CXX_COMPILER}"
+      python3 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py -v --api-only
     BYPRODUCTS
       ${_cybld}/thrift/python/types_api.h
       ${_cybld}/thrift/python/server/server_api.h
@@ -351,26 +352,29 @@ if(thriftpy3)
   endif()
 
   add_custom_command(TARGET thrift_python_and_py3_bindings
-    COMMAND
-      CXX="${CMAKE_CXX_COMPILER}"
-      LDSHARED="${CMAKE_CXX_COMPILER} -shared"
-      python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py
+    COMMAND ${CMAKE_COMMAND} -E env
+      "CFLAGS=${CMAKE_C_FLAGS}"
+      "CXXFLAGS=${CMAKE_CXX_FLAGS}"
+      "CXX=${CMAKE_CXX_COMPILER}"
+      python3 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py -v
       build_ext -f ${incs} ${libs} --libpython ${python_libname}
     WORKING_DIRECTORY ${_cybld}
   )
   add_custom_command(TARGET thrift_python_and_py3_bindings
-    COMMAND
-      CXX="${CMAKE_CXX_COMPILER}"
-      LDSHARED="${CMAKE_CXX_COMPILER} -shared"
-      python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py
+    COMMAND ${CMAKE_COMMAND} -E env
+      "CFLAGS=${CMAKE_C_FLAGS}"
+      "CXXFLAGS=${CMAKE_CXX_FLAGS}"
+      "CXX=${CMAKE_CXX_COMPILER}"
+      python3 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py -v
       bdist_wheel --libpython ${python_libname}
     WORKING_DIRECTORY ${_cybld}
   )
   add_custom_command(TARGET thrift_python_and_py3_bindings POST_BUILD
-    COMMAND
-      CXX="${CMAKE_CXX_COMPILER}"
-      LDSHARED="${CMAKE_CXX_COMPILER} -shared"
-      python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py
+    COMMAND ${CMAKE_COMMAND} -E env
+      "CFLAGS=${CMAKE_C_FLAGS}"
+      "CXXFLAGS=${CMAKE_CXX_FLAGS}"
+      "CXX=${CMAKE_CXX_COMPILER}"
+      python3 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py -v
       install --prefix ${CMAKE_INSTALL_PREFIX} --libpython ${python_libname}
     WORKING_DIRECTORY ${_cybld}
   )
@@ -381,8 +385,10 @@ if(thriftpy3)
   set(ENV{CXX} ${CMAKE_CXX_COMPILER})
   install(CODE "
     execute_process(
-      COMMAND
-        python ${CMAKE_CURRENT_SOURCE_DIR}/setup.py install
+      COMMAND ${CMAKE_COMMAND} -E env
+        \"CFLAGS=${CMAKE_C_FLAGS}\"
+        \"CXXFLAGS=${CMAKE_CXX_FLAGS}\"
+        python3 ${CMAKE_CURRENT_SOURCE_DIR}/setup.py install -v
         --prefix ${py_install_dir} --libpython ${python_libname}
       COMMAND_ECHO STDOUT
       WORKING_DIRECTORY ${_cybld}
diff --git a/thrift/lib/cpp2/CMakeLists.txt b/thrift/lib/cpp2/CMakeLists.txt
index 5c87e73a46..2403422334 100644
--- a/thrift/lib/cpp2/CMakeLists.txt
+++ b/thrift/lib/cpp2/CMakeLists.txt
@@ -16,9 +16,9 @@
 set(LIB_CPP2_HOME ${CMAKE_CURRENT_SOURCE_DIR})
 
 if (enable_tests)
-  add_subdirectory(protocol/test)
+#  add_subdirectory(protocol/test)
+#  add_subdirectory(test)
 endif ()
-add_subdirectory(test)
 
 #######
 # CMAKE variables only have local/subdirectory scope
diff --git a/thrift/lib/setup.py b/thrift/lib/setup.py
index 3f2e5beea3..87dfd8e5f9 100644
--- a/thrift/lib/setup.py
+++ b/thrift/lib/setup.py
@@ -18,6 +18,7 @@
 # created as tree of links to the real source files.
 
 import sys
+import os
 
 import Cython
 from Cython.Build import cythonize
@@ -82,7 +83,6 @@ else:
     common_options = {
         "language": "c++",
         "libraries": libs,
-        "extra_compile_args": ["-std=gnu++17", "-fcoroutines"],
     }
 
     exts = [
